src/ui/components/Table/Table.tsx

```diff
@@ -0,0 +1,321 @@
+    const {head = [], rows = []} = args;
+    const groupedCells = new Map();
+
+    let cellIndex = 0;
+    const setGrouping = (col: THead) => {
+        if (col.columns?.length) {
+            col.columns.forEach(setGrouping);
+        } else {
+            if (col.enableRowGrouping) {
+                groupedCells.set(cellIndex, {rowIndex: -1});
+            }
+            cellIndex++;
+        }
+    };
+
+    head.forEach(setGrouping);
+
+    if (groupedCells.size) {
+        return rows.map((row, rowIndex) => {
+            return row.map((currentCell, index) => {
+                if (groupedCells.has(index)) {
+                    const prevCellData = groupedCells.get(index);
+                    const prevCell = rows[prevCellData.rowIndex]?.[index];
+
+                    if (prevCell && currentCell.value === prevCell?.value) {
+                        prevCell.rowSpan = (prevCell.rowSpan || 1) + 1;
+                        return {...currentCell, isVisible: false};
+                    } else {
+                        groupedCells.set(index, {rowIndex});
+                    }
+                }
+
+                return currentCell;
+            });
+        });
+    }
+
+    return rows;
+}
+
+export const Table = (props: TableProps) => {
+    const {
+        title,
+        noData,
+        onClick,
+        header: headerOptions,
+        qa,
+        manualSorting,
+        onSortingChange,
+    } = props;
+    const {head, footer, rows} = props.data;
+    const columns = React.useMemo(() => getTableColumns({head, footer}), [head, footer]);
+    const data = React.useMemo(() => getTableData({head, rows}), [head, rows]);
+    const [sorting, setSorting] = React.useState<SortingState>([]);
+    const table = useReactTable({
+        data,
+        columns,
+        getCoreRowModel: getCoreRowModel(),
+        getSortedRowModel: getSortedRowModel(),
+        getGroupedRowModel: getGroupedRowModel(),
+        sortDescFirst: false,
+        manualSorting,
+        state: {
+            sorting,
+        },
+        onSortingChange: (updater) => {
+            setSorting(updater);
+
+            if (onSortingChange) {
+                const updates = typeof updater === 'function' ? updater(sorting) : updater;
+                const id = updates[0]?.id;
+                const desc = updates[0]?.desc;
+                const headCellData = columns.find((c) => c.id === id)?.meta?.head;
+                onSortingChange({cell: headCellData, sortOrder: desc ? 'desc' : 'asc'});
+            }
+        },
+    });
+
+    const handleCellClick: OnTableClick = (args) => {
+        if (onClick) {
+            onClick(args);
+        }
+    };
+
+    const tableRows = table.getRowModel().rows;
+    const tableHeaders = table.getHeaderGroups();
+    const shouldShowFooter = columns.some((column) => column.footer);
+
+    return (
+        <div className={b()}>
+            {title && <div className={b('title')}>{title.text}</div>}
```

Comment:
Maybe we should use [native caption](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/caption)? Then we will need no more a `div` wrapper